src/main/java/com/znsio/e2e/runner/Setup.java

```diff
@@ -354,34 +358,41 @@ private void checkIfAppExistsAtTheMentionedPath (String appPath, String capabili
     }
 
     private void fetchWindowsAppVersion() {
+        Pattern VERSION_NAME_PATTERN = Pattern.compile("Version=([0-9]+(\\.[0-9]+)+)", Pattern.MULTILINE);
+
         String appPath = String.valueOf(configs.get(APP_PATH));
         String nameVariable = "name=\"" + appPath.replace("\\", "\\\\") + "\"";
         String[] commandToGetAppVersion = new String[] {"wmic", "datafile", "where", nameVariable, "get", "Version", "/value"};
-        fetchAppVersion(commandToGetAppVersion);
+        fetchAppVersion(commandToGetAppVersion, VERSION_NAME_PATTERN);
     }
 
     private void fetchAndroidAppVersion() {
-        String[] commandToGetAppVersion;
-        if ((null == devices) || (devices.isEmpty())) {
-            LOGGER.info("fetchAndroidAppVersion: devices list found empty!");
-            commandToGetAppVersion = new String[] {"adb", "shell", "dumpsys", "package", configs.get(APP_PACKAGE_NAME), "|", "findstr", "versionName"};
+        Pattern VERSION_NAME_PATTERN = Pattern.compile("versionName='([0-9]+(\\.[0-9]+)+)'", Pattern.MULTILINE);
+        String filePath = configs.get(APP_PATH);
+
+        String env = System.getenv("ANDROID_HOME");
+        File buildToolsFolder = new File(env, "build-tools");
+        File buildTool = buildToolsFolder.listFiles()[0];
+        File[] aaptFilter = buildTool.listFiles((dir, name) -> name.toLowerCase().contains("aapt.exe"));
```

Comment:
What if you are running on OSX / Linux - we will not have appt.exe there.

Reply:
made the required changes